packages/waku/src/lib/plugins/vite-plugin-rsc-index.ts

```diff
@@ -26,9 +26,9 @@ ${config.htmlHead}
     configureServer(server) {
       return () => {
         server.middlewares.use((req, res) => {
-          res.statusCode = 200;
-          res.setHeader('content-type', 'text/html; charset=utf-8');
           server.transformIndexHtml(req.url || '', html).then((content) => {
+            res.statusCode = 200;
```

Comment:
This is not related to the PR, but it's a minor change, since we used to set the status 200 even though the `.then` might never work and the promise might reject. 

So we just bring it when we're sure the promise is getting resolved.